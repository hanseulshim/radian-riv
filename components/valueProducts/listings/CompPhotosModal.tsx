import React from 'react'
import Modal from 'components/common/Modal'
import { PropertyInterface } from 'api'
import CompPhoto from './CompPhoto'
import Checkbox from 'components/common/Checkbox'

interface Props {
  closeModal: () => void
  setCheckedProperties: React.Dispatch<React.SetStateAction<string[]>>
  checkedProperties: string[]
  properties: PropertyInterface[]
  view: string
}

export default function CompPhotosModal({
  closeModal,
  view,
  properties,
  checkedProperties,
  setCheckedProperties
}: Props) {
  return (
    <Modal closeModal={closeModal} title={''} percent={90} id="comp-modal">
      <div className="comp-photos-modal">
        <div className="top-row">
          <button className="btn btn-small">Submit</button>
          <div className="check">
            <Checkbox
              label="Check/Uncheck All"
              checked={checkedProperties.length === properties.length}
              onChange={() => {
                if (!(checkedProperties.length === properties.length)) {
                  const listings = properties.map(
                    property => property.listingNumber
                  )
                  setCheckedProperties(listings)
                } else {
                  setCheckedProperties([])
                }
              }}
              leftLabel
            />
          </div>
        </div>
        <div className="comp-photo-container">
          {properties.map(property => (
            <CompPhoto
              key={property.listingNumber}
              property={property}
              setCheckedProperties={setCheckedProperties}
              checkedProperties={checkedProperties}
              view={view}
            />
          ))}
        </div>
      </div>
      <button className="btn btn-small">Submit</button>
    </Modal>
  )
}
