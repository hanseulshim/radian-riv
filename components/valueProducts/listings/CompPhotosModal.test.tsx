import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CompPhotosModal from './CompPhotosModal'

describe('CompPhotosModal', () => {
  let component
  beforeAll(() => {
    const props = {
      closeModal: () => {},
      setCheckedProperties: () => {},
      checkedProperties: [],
      view: 'test',
      properties: [
        {
          actDom: 204,
          address: '684 Willms Roads',
          agSqft: 4596,
          basement: 'Full',
          bath: 2,
          bed: 5,
          city: 'East Dean',
          coeDate: '6/25/2020',
          concessions: 16464,
          description:
            'Esse commodi et eum voluptatem rerum. Non rerum quia. Quia nisi vel minus cumque neque reprehenderit architecto et dolores. Nostrum in maiores cum. Quia ut sed omnis error cumque necessitatibus perferendis esse dolor. Provident porro aperiam dolorem dolores.',
          distressed: true,
          garage: 2,
          listDate: '12/27/2020',
          listPrice: 552470,
          listPricePerSqft: 3998,
          listingNumber: 'MDMC662271',
          listingSheetSelected: false,
          lotSize: 1,
          order: 1,
          photos: [],
          pool: 'None',
          schoolDistrict: 'Borders County Public Schools',
          soldPrice: 585277,
          sqft: 3355,
          sqftPrice: 145.15,
          subdivision: 'Fresh Chair Village',
          targetDistance: 0.68,
          totDom: 172,
          totalSqft: 4243,
          valuationPercent: 0.62,
          waterfront: 'Water Oriented: No',
          yearBuilt: 2010,
          zip: '03194-7596'
        }
      ]
    }
    component = shallow(<CompPhotosModal {...props} />)
  })
  it('has comp-modal container', () => {
    expect(component.get('#comp-modal'))
    expect(component.get('.comp-photos-modal'))
    expect(component.get('.top-row'))
    expect(component.get('.comp-photo-container'))
  })
  it('two submit buttons', () => {
    expect(component.find('.btn').length).toEqual(2)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
