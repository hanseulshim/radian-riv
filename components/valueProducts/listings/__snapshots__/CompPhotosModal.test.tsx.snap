// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompPhotosModal should match snapshot 1`] = `
<Modal
  closeModal={[Function]}
  id="comp-modal"
  percent={90}
  title=""
>
  <div
    className="comp-photos-modal"
  >
    <div
      className="top-row"
    >
      <button
        className="btn btn-small"
      >
        Submit
      </button>
      <div
        className="check"
      >
        <Checkbox
          checked={false}
          label="Check/Uncheck All"
          leftLabel={true}
          onChange={[Function]}
        />
      </div>
    </div>
    <div
      className="comp-photo-container"
    >
      <CompPhoto
        checkedProperties={Array []}
        key="MDMC662271"
        property={
          Object {
            "actDom": 204,
            "address": "684 Willms Roads",
            "agSqft": 4596,
            "basement": "Full",
            "bath": 2,
            "bed": 5,
            "city": "East Dean",
            "coeDate": "6/25/2020",
            "concessions": 16464,
            "description": "Esse commodi et eum voluptatem rerum. Non rerum quia. Quia nisi vel minus cumque neque reprehenderit architecto et dolores. Nostrum in maiores cum. Quia ut sed omnis error cumque necessitatibus perferendis esse dolor. Provident porro aperiam dolorem dolores.",
            "distressed": true,
            "garage": 2,
            "listDate": "12/27/2020",
            "listPrice": 552470,
            "listPricePerSqft": 3998,
            "listingNumber": "MDMC662271",
            "listingSheetSelected": false,
            "lotSize": 1,
            "order": 1,
            "photos": Array [],
            "pool": "None",
            "schoolDistrict": "Borders County Public Schools",
            "soldPrice": 585277,
            "sqft": 3355,
            "sqftPrice": 145.15,
            "subdivision": "Fresh Chair Village",
            "targetDistance": 0.68,
            "totDom": 172,
            "totalSqft": 4243,
            "valuationPercent": 0.62,
            "waterfront": "Water Oriented: No",
            "yearBuilt": 2010,
            "zip": "03194-7596",
          }
        }
        setCheckedProperties={[Function]}
        view="test"
      />
    </div>
  </div>
  <button
    className="btn btn-small"
  >
    Submit
  </button>
</Modal>
`;
